### PR TITLE
Round.IgnoredPlayers  fix

### PR DESCRIPTION
**Pull request by louis1706**

### CODE REVIEW

File: Exiled.Events/Patches/Events/Server/RoundEnd.cs [hunk 1]
```diff
@@ -46,18 +45,40 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
             const string LeadingTeam = "<leadingTeam>5__9";
             const string NewList = "<newList>5__3";
 
+            int offset = -1;
+            int index = newInstructions.FindIndex(x => x.Calls(Method(typeof(PlayerRolesUtils), nameof(PlayerRolesUtils.GetTeam), new Type[] { typeof(ReferenceHub), }))) + offset;
+
+            Label labelcontinu = generator.DefineLabel();
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Server/RoundEnd.cs:51]:
```suggestion
            Label continue = generator.DefineLabel();
```

File: Exiled.Events/Patches/Events/Server/RoundEnd.cs [hunk 2]
```diff
@@ -46,18 +45,40 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
...
+            // if (Round.IgnoredPlayers.Contains(referencehub)
+            //  goto labelcontinu;
+            newInstructions.InsertRange(
+                index,
+                new CodeInstruction[]
+                {
+                    new(OpCodes.Call, PropertyGetter(typeof(Round), nameof(Round.IgnoredPlayers))),
+                    new(OpCodes.Ldloc_S, 10),
+                    new(OpCodes.Call, Method(typeof(HashSet<ReferenceHub>), nameof(HashSet<ReferenceHub>.Contains))),
+                    new(OpCodes.Brtrue_S, labelcontinu),
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Server/RoundEnd.cs:62]:
```suggestion
                    new(OpCodes.Brtrue_S, continue),
```

File: Exiled.Events/Patches/Events/Server/RoundEnd.cs [hunk 3]
```diff
@@ -46,18 +45,40 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
             const string LeadingTeam = "<leadingTeam>5__9";
             const string NewList = "<newList>5__3";
 
+            int offset = -1;
+            int index = newInstructions.FindIndex(x => x.Calls(Method(typeof(PlayerRolesUtils), nameof(PlayerRolesUtils.GetTeam), new Type[] { typeof(ReferenceHub), }))) + offset;
+
+            Label labelcontinu = generator.DefineLabel();
+
+            // if (Round.IgnoredPlayers.Contains(referencehub)
+            //  goto labelcontinu;
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Server/RoundEnd.cs:54]:
```suggestion
            //  goto continue;
```

File: Exiled.Events/Patches/Events/Server/RoundEnd.cs [hunk 4]
```diff
@@ -46,18 +45,40 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
...
+                    new(OpCodes.Call, PropertyGetter(typeof(Round), nameof(Round.IgnoredPlayers))),
+                    new(OpCodes.Ldloc_S, 10),
+                    new(OpCodes.Call, Method(typeof(HashSet<ReferenceHub>), nameof(HashSet<ReferenceHub>.Contains))),
+                    new(OpCodes.Brtrue_S, labelcontinu),
+                });
+
+            offset = 4;
+            index = newInstructions.FindIndex(x => x.Calls(Method(typeof(PlayerRolesUtils), nameof(PlayerRolesUtils.GetTeam), new Type[] { typeof(ReferenceHub), }))) + offset;
+
+            newInstructions[index].labels.Add(labelcontinu);
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Server/RoundEnd.cs:68]:
```suggestion
            newInstructions[index].labels.Add(continue);
```

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

File: Exiled.Events/Patches/Events/Server/RoundEnd.cs (hunk 1 above)
Comment (louis1706) [reply Exiled.Events/Patches/Events/Server/RoundEnd.cs:51]: you can't name a variable continue or return

Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Server/RoundEnd.cs:51]:
```suggestion
            Label jmp = generator.DefineLabel();
```

File: Exiled.Events/Patches/Events/Server/RoundEnd.cs [hunk 5]
```diff
@@ -46,18 +45,40 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
...
+            // if (Round.IgnoredPlayers.Contains(referencehub)
+            //  goto labelcontinu;
+            newInstructions.InsertRange(
+                index,
+                new CodeInstruction[]
+                {
+                    new(OpCodes.Call, PropertyGetter(typeof(Round), nameof(Round.IgnoredPlayers))),
+                    new(OpCodes.Ldloc_S, 10),
+                    new(OpCodes.Call, Method(typeof(HashSet<ReferenceHub>), nameof(HashSet<ReferenceHub>.Contains))),
+                    new(OpCodes.Brtrue_S, labelcontinu),
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Server/RoundEnd.cs:62]:
```suggestion
                    new(OpCodes.Brtrue_S, jmp),
```

File: Exiled.Events/Patches/Events/Server/RoundEnd.cs [hunk 6]
```diff
@@ -46,18 +45,40 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
             const string LeadingTeam = "<leadingTeam>5__9";
             const string NewList = "<newList>5__3";
 
+            int offset = -1;
+            int index = newInstructions.FindIndex(x => x.Calls(Method(typeof(PlayerRolesUtils), nameof(PlayerRolesUtils.GetTeam), new Type[] { typeof(ReferenceHub), }))) + offset;
+
+            Label jmp = generator.DefineLabel();
+
+            // if (Round.IgnoredPlayers.Contains(referencehub)
+            //  goto labelcontinu;
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Server/RoundEnd.cs:54]:
```suggestion
            //  goto jmp;
```

File: Exiled.Events/Patches/Events/Server/RoundEnd.cs [hunk 7]
```diff
@@ -46,18 +45,40 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
...
+                    new(OpCodes.Call, PropertyGetter(typeof(Round), nameof(Round.IgnoredPlayers))),
+                    new(OpCodes.Ldloc_S, 10),
+                    new(OpCodes.Call, Method(typeof(HashSet<ReferenceHub>), nameof(HashSet<ReferenceHub>.Contains))),
+                    new(OpCodes.Brtrue_S, labelcontinu),
+                });
+
+            offset = 4;
+            index = newInstructions.FindIndex(x => x.Calls(Method(typeof(PlayerRolesUtils), nameof(PlayerRolesUtils.GetTeam), new Type[] { typeof(ReferenceHub), }))) + offset;
+
+            newInstructions[index].labels.Add(labelcontinu);
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Server/RoundEnd.cs:68]:
```suggestion
            newInstructions[index].labels.Add(jmp);
```

**Review by NaoUnderscore — COMMENTED** [review]
fuck long names

**Pull request merged**